sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -123,13 +124,23 @@ public Long addSample(final Sample sample) throws IOException {
   }
 
   @Override
-  public int getNextSiblingNumber(Sample parent, SampleClass childClass) throws IOException {
-    Query query = currentSession().createQuery(
-        "select max(siblingNumber) " + "from DetailedSampleImpl " + "where parentId = :parentId " + "and sampleClassId = :sampleClassId");
-    query.setLong("parentId", parent.getId());
-    query.setLong("sampleClassId", childClass.getId());
+  public int getNextSiblingNumber(String partialAlias) throws IOException {
+    Query query = currentSession().createQuery("select max(siblingNumber) from DetailedSampleImpl as ds"
+            + " where alias IN (concat(:alias, ds.siblingNumber), concat(:alias, '0', ds.siblingNumber))");
+    query.setString("alias", partialAlias);
     Number result = ((Number) query.uniqueResult());
-    return result == null ? 1 : result.intValue() + 1;
+    int next = result == null ? 0 : result.intValue();
+
+    // verify uniqueness and fix siblingNumbers for existing samples if necessary
```

Comment:
This could use some explanation.

First, the scope of a sibling number is a tissue, right?

As I understand it, you find the highest sibling number that presently exists and then set the sibling number of the all the samples that share that prefix (i.e., the same tissue?) and any of their descendants who already have a higher sibling number? Yes?